src/core/webview/ClineProvider.ts

```diff
@@ -1169,14 +1176,28 @@ export class ClineProvider implements vscode.WebviewViewProvider {
 							vscode.window.showErrorMessage("Failed to get list api configuration")
 						}
 						break
-					case "experimentalDiffStrategy":
-						await this.updateGlobalState("experimentalDiffStrategy", message.bool ?? false)
+					case "updateExperimental": {
+						if (!message.values) {
+							break
+						}
+
+						const updatedExperiments = {
+							...((await this.getGlobalState("experiments")) ?? experimentDefault),
+							...message.values,
+						}
+
+						await this.updateGlobalState("experiments", updatedExperiments)
+
 						// Update diffStrategy in current Cline instance if it exists
-						if (this.cline) {
-							await this.cline.updateDiffStrategy(message.bool ?? false)
+						if (message.values[EXPERIMENT_IDS.DIFF_STRATEGY] !== undefined && this.cline) {
+							await this.cline.updateDiffStrategy(
+								Experiments.isEnabled(updatedExperiments, EXPERIMENT_IDS.DIFF_STRATEGY),
+							)
 						}
```

Comment:
In general I'm curious why this doesn't just look at the extension state like the other settings